app/src/main/java/com/example/bookwormadventuresdeluxe2/Book.java

```diff
@@ -38,6 +39,8 @@ public Book(String owner, String title, String author, String description, Strin
         this.isbn = isbn;
         this.description = description;
         this.status = status;
+        this.pickUpAddress = pickUpAddress;
+        this.requesters = requesters;
     }

```

Comment:
This constructor is good for testing, but for implementation we will always need to pass empty list and address.
Should probably add a second constructor for general use (set requester, borrower, address null / empty)

Also this missed the borrower?

Reply:
Is this what you meant by null?. Because the adapter goes from -document to Book- for all books right away so we need a working constructor if we want to pick up empty address/empty string and filled address/filled string
![image](https://user-images.githubusercontent.com/39696855/97760188-2f0b8e80-1ac8-11eb-94b4-2fc8ef06b422.png)

ex//
Book1
owner
title
author
...
address: ""
requesters: [""]

Book2
owner
title
author
address: 10101 101 Street
requesters: [bob]

book1.getAddress = ""
book2.getAddress = 10101 101 Street

If you set it to null the -document to Book- query wont work if some books have two null fields and some dont

Sorry im confused a bit. Did you want it work the same was as having an empty description "" in the constructor
------------
Also what do I do about the borrower also, do I make the book constructor just  "" or null?
Book(owner, title, author, isbn, description, status)
or
Book(owner, title, author, isbn, description, status, borrower = "")

Reply:
idk why that is bolded^ i tried adding a --- or *** or something

Reply:
I just wanted a constructor for new books that auto sets values so I dont have to go
new Book(owner, title, author, description, isbn,  Status.Available, "",  new ArrayList<String>())
but instead can call
new Book(owner, title, author, description, isbn)
and have the constructor do the rest (it is a new book, status = Available, address = empty, borrower = empty, requesters = empty list) by default